tests/Query.test.ts

```diff
@@ -322,6 +451,95 @@ describe('Query', () => {
         );
     });
 
+    it('filters based off a single tag', () => {
+        // Arrange
+        const originalSettings = getSettings();
+        updateSettings({ globalFilter: '#task' });
+        const tasks: Task[] = [
+            Task.fromLine({
+                line: '- [ ] #task something to do #later #work 📅 2021-09-12 ✅ 2021-06-20',
+                sectionStart: 0,
+                sectionIndex: 0,
+                path: '',
+                precedingHeader: '',
+            }),
+            Task.fromLine({
+                line: '- [ ] #task something to do #later #home 📅 2021-09-12 ✅ 2021-06-20',
+                sectionStart: 0,
+                sectionIndex: 0,
+                path: '',
+                precedingHeader: '',
+            }),
+            Task.fromLine({
+                line: '- [ ] #task get the milk 📅 2021-09-12 ✅ 2021-06-20',
+                sectionStart: 0,
+                sectionIndex: 0,
+                path: '',
+                precedingHeader: '',
+            }),
+        ] as Task[];
+        const input = 'tags include #home';
+        const query = new Query({ source: input });
+
+        // Act
+        let filteredTasks = [...tasks];
+        query.filters.forEach((filter) => {
+            filteredTasks = filteredTasks.filter(filter);
+        });
+
+        // Assert
+        expect(filteredTasks.length).toEqual(1);
+        expect(filteredTasks[0]).toEqual(tasks[1]);
+
+        // Cleanup
+        updateSettings(originalSettings);
```

Comment:
@sytone it would be nice to explain the reasoning behind the test cases a bit. I don't think this should prevent us from merging, though @claremacrae.

Reply:
Sure.